src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -210,13 +210,23 @@ protected override ConstructorBuilder DefineDefaultConstructorCore(MethodAttribu
         private ConstructorBuilderImpl DefineDefaultConstructorInternal(MethodAttributes attributes)
         {
             // Get the parent class's default constructor and add it to the IL
-            ConstructorInfo? con;
-            if (_typeParent!.IsConstructedGenericType && _typeParent.GetGenericTypeDefinition() is TypeBuilderImpl typeBuilder)
+            ConstructorInfo? con = null;
+            if (_typeParent!.IsConstructedGenericType)
             {
-                con = GetConstructor(_typeParent, typeBuilder.GetConstructor(
-                    BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, EmptyTypes, null)!);
+                Type typeDefinition = _typeParent.GetGenericTypeDefinition();
+                if (typeDefinition is TypeBuilderImpl typeBuilder)
+                {
+                    con = GetConstructor(_typeParent, typeBuilder.GetConstructor(
+                        BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, EmptyTypes, null)!);
+                }
+                else if (ModuleBuilderImpl.ContainsTypeBuilder(_typeParent.GetGenericArguments()))
+                {
+                    con = GetConstructor(_typeParent, typeDefinition.GetConstructor(
```

Comment:
There is only a subtle difference (`typeBuilder` vs `typeDefinition`) between these two branches. Can you add a comment on the decision we are making here and why it is appropriate?

Reply:
Ah right, it was actually same, no need branching out